File: laba2/lab2.kt
fun searchColor(r: String): Int {
    if (r.length % 2 != 0) {
        println("Incorrect value")
        return 0
    }

    val vecR = IntArray(10)
    val vecW = IntArray(10)
    val vecB = IntArray(10)
    var c = 0

    for (i in r.indices step 2) {
        val color = r[i]
        val cod = r[i + 1] - '0'
        when (color) {
            'R' -> vecR[cod] = 1
            'W' -> vecW[cod] = 1
            'B' -> vecB[cod] = 1
        }
    }

    for (i in 0..9) {
        if (vecR[i] == 1 && vecW[i] == 1 && vecB[i] == 1) c++
    }

    return c
}

fun main() {
    print("Enter rods: ")
    val input = readLine()!!
    println("Number of rods: ${searchColor(input)}")
}

https://www.tutorialspoint.com/compilers/online-kotlin-compiler.htm

File: laba2/lab2.js
function searchColor(r) {
    if (r.length % 2 !== 0) {
        console.log("Incorrect value");
        return 0;
    }

    let c = 0;
    const vecR = Array(10).fill(0);
    const vecW = Array(10).fill(0);
    const vecB = Array(10).fill(0);

    for (let i = 0; i < r.length; i += 2) {
        const color = r[i];
        const cod = parseInt(r[i + 1]);

        if (color === 'R') vecR[cod] = 1;
        else if (color === 'W') vecW[cod] = 1;
        else if (color === 'B') vecB[cod] = 1;
    }

    for (let i = 0; i < 10; i++) {
        if (vecR[i] && vecW[i] && vecB[i]) c++;
    }

    return c;
}

const input = prompt("Enter rods:");
console.log("Number of rods:", searchColor(input));
